File: insfrastructure/entities/person_entity.py
from uuid import uuid4

class PersonEntity:
    def __init__(
            self,
            # id: uuid4(), 
            first_name: str, 
            last_name: str, 
            date_of_birth: str, 
            email: str, 
            national_id: str, 
            phone_number: str = 'N/A', 
            parmanent_address: str = 'N/A', 
            gender: str = 'N/A',
        ):
        self.id = uuid4()
        self.first_name = first_name
        self.last_name = last_name
        self.date_of_birth = date_of_birth
        self.gender = gender
        self.email = email
        self.national_id = national_id
        self.parmanent_address = parmanent_address
        self.phone_number = phone_number

File: typescript-impl/src/infrastructure/entities/person_entity.ts
import { v4 as uuidv4 } from 'uuid';
import { Gender } from '../enums/gender_choice';

export class PersonEntity {
    id: string;
    first_name: string;
    last_name: string;
    date_of_birth: Date;
    gender: Gender;
    email: string;
    national_id: string;
    permanent_address: string;
    phone_number: string;

    constructor(
        first_name: string,
        last_name: string,
        date_of_birth: Date,
        email: string,
        gender: Gender,
        national_id: string,
        phone_number: string,
        permanent_address: string,
    ) {
        this.id = uuidv4();
        this.first_name = first_name;
        this.last_name = last_name;
        this.date_of_birth = date_of_birth;
        this.gender = gender;
        this.email = email;
        this.national_id = national_id;
        this.permanent_address = permanent_address;
        this.phone_number = phone_number;
    }
}
